beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -687,3 +705,18 @@ func (f *ForkChoice) Weight(root [32]byte) (uint64, error) {
 	}
 	return n.weight, nil
 }
+
+// updateJustifiedBalances updates the validators balances on the justified checkpoint pointed by root
+func (f *ForkChoice) updateJustifiedBalances(ctx context.Context, root [32]byte) error {
+	balances, err := f.balancesByRoot(ctx, root)
+	if err != nil {
+		return errors.Wrap(err, "could not get justified balances")
+	}
+	f.justifiedBalances = balances
+	return nil
+}
+
+// SetJustifiedBalances is used only in tests
+func (f *ForkChoice) SetJustifiedBalances(bals []uint64) {
+	f.justifiedBalances = bals
+}
```

Comment:
Didn't Preston recommend some tricks to hide behind such methods behind tests only setup?

Reply:
For non-pure functions is a little awkward

Reply:
Convert it to a pure function with an additional `*ForkChoice` parameter. Having such a setter is dangerous.